vendor/mosn.io/api/xprotocol.go

```diff
@@ -143,6 +144,13 @@ type XProtocol interface {
 	GenerateRequestID(*uint64) uint64
 }
 
+// GracefulShutdown provides the ability to construct proper GoAway command for xprotocol,
+// it's not required, please NOT implement this interface if the protocol doesn't has goaway.
+type GracefulShutdown interface {
+	// GoAway builds an active GoAway command, requestId is the maxClientStreamID
+	GoAway(context context.Context, requestId uint64) XFrame
```

Comment:
Is it better scalability to support goaway and goaway-ack also?

Reply:
I think it's better to keep it simple at first.
Http2.0 and quick do not have goaway ack frame.